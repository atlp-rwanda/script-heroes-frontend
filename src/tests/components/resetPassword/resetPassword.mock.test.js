import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import ResetPassword from "../../../components/resetPassword/resetPassword";
import Adapter from "enzyme-adapter-react-16";
import configureStore from "redux-mock-store";
import {useSelector, useDispatch} from 'react-redux'
import { useParams } from "react-router-dom";
import { act } from "react-dom/test-utils";

const mockDispatch = jest.fn();
const mockSelector = jest.fn();
const mockParams = jest.fn();

jest.mock("react-redux", () => ({
  useSelector: () => mockSelector,
  useDispatch: () => mockDispatch,
}));
jest.mock("react-router-dom", () => ({
  useParams: () => mockParams,
}));

describe("reset page", () => {
  let wrapper;
  beforeEach(() => {
    act(() => {
      wrapper = mount(<ResetPassword />);
    });
  });
  it("Should render reset with no errors", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
