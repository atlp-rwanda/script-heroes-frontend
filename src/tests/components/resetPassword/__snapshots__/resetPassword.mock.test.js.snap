// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reset page Should render reset with no errors 1`] = `"<div class=\\"container-fluid\\"><div class=\\"resetPassword\\"><div class=\\"row\\"><div class=\\"col-md-6\\"><form id=\\"resetForm\\" class=\\"\\"><center><h1>Reset Password</h1></center><div class=\\"resetContainer\\"><label class=\\"\\">New Password</label><br><div class=\\"passwordInput\\"><input placeholder=\\"Enter your new password\\" type=\\"password\\" class=\\"form-control\\" value=\\"\\"><i><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"eye\\" class=\\"svg-inline--fa fa-eye fa-w-18 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 576 512\\"><path fill=\\"currentColor\\" d=\\"M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z\\"></path></svg></i></div><br><label>Confirm Password</label><br><div class=\\"passwordInput\\"><input placeholder=\\"Repeat your password\\" type=\\"password\\" class=\\"form-control\\" value=\\"\\"><i id=\\"myEye\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"eye\\" class=\\"svg-inline--fa fa-eye fa-w-18 \\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 576 512\\"><path fill=\\"currentColor\\" d=\\"M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z\\"></path></svg></i></div><center><div id=\\"response\\"><div role=\\"status\\" class=\\"spinner-border text-primary\\"><span class=\\"sr-only\\">Loading...</span></div></div><div class=\\"alert alert-success fade\\" role=\\"alert\\"></div><button type=\\"submit\\" id=\\"resetBtn\\" class=\\"btn btn-secondary\\">Reset</button></center></div></form></div><div class=\\"col-md-6\\"><img class=\\"resetImg\\" src=\\"test-file-stub\\" alt=\\"image\\"></div></div></div></div>"`;
